File: Easy/esonero/meta_esonerotitolostudioview/meta_esonerotitolostudioview.cs

/*
Easy
Copyright (C) 2021 UniversitÓ degli Studi di Catania (www.unict.it)
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/


using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_esonerotitolostudioview
{
    public class Meta_esonerotitolostudioview : Meta_easydata 
	{
        public Meta_esonerotitolostudioview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "esonerotitolostudioview") {
				Name = "Definizione degli esoneri per titoli di studio conseguiti";
			EditTypes.Add("titolostudio");
			ListingTypes.Add("titolostudio");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idesonero"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "titolostudio": {
						DescribeAColumn(T, "aa", "Anno accademico", nPos++);
						DescribeAColumn(T, "title", "Denominazione", nPos++);
						DescribeAColumn(T, "esonero_description", "Descrizione", nPos++);
						DescribeAColumn(T, "esonero_applunavolta", "Applicabile una sola volta", nPos++);
						DescribeAColumn(T, "costoscontodef_title", "Sconto", nPos++);
						DescribeAColumn(T, "esoneroanskind_title", "Tipologia Codice ANS", nPos++);
						DescribeAColumn(T, "esoneroanskind_description", "Descrizione Codice ANS", nPos++);
						DescribeAColumn(T, "esonero_retroattivo", "Retroattivo", nPos++);
						DescribeAColumn(T, "esonero_soloconisee", "Applicabile solo con ISEE", nPos++);
						DescribeAColumn(T, "esonero_titolostudio_conseguitoincorso", "Conseguito in corso", nPos++);
						DescribeAColumn(T, "esonero_titolostudio_dataconstutticf", "Data limite per aver conseguito tutti i crediti formativi", nPos++);
						DescribeAColumn(T, "esonero_titolostudio_datalaurea", "Data limite di conseguimento del titolo", nPos++);
						DescribeAColumn(T, "struttura_title", "Struttura didattica", nPos++);
						DescribeAColumn(T, "esonero_titolostudio_nellistituto", "Solo per corsi dell'istituto", nPos++);
						DescribeAColumn(T, "esonero_titolostudio_noabbr", "Senza abbreviazioni di carriera", nPos++);
						DescribeAColumn(T, "esonero_titolostudio_noparttime", "Senza aver effettuato iscrizioni part-time", nPos++);
						DescribeAColumn(T, "esonero_titolostudio_votomin", "Voto minimo", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "titolostudio": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/esonero/meta_esonerotitolostudioview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_esonerotitolostudioview() {
        MetaData.apply(this, ["esonerotitolostudioview"]);
        this.name = 'meta_esonerotitolostudioview';
    }

    meta_esonerotitolostudioview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_esonerotitolostudioview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'titolostudio':
						this.describeAColumn(table, 'aa', 'Anno accademico', null, 1000, 9);
						this.describeAColumn(table, 'esonero_titolostudio_conseguitoincorso', 'Conseguito in corso', null, 1000, null);
						this.describeAColumn(table, 'esonero_titolostudio_dataconstutticf', 'Data limite per aver conseguito tutti i crediti formativi', null, 2000, null);
						this.describeAColumn(table, 'title', 'Denominazione', null, 3000, 50);
						this.describeAColumn(table, 'esonero_titolostudio_datalaurea', 'Data limite di conseguimento del titolo', null, 3000, null);
						this.describeAColumn(table, 'esonero_description', 'Descrizione', null, 4000, 256);
						this.describeAColumn(table, 'esonero_applunavolta', 'Applicabile una sola volta', null, 5000, null);
						this.describeAColumn(table, 'struttura_title', 'Denominazione Struttura didattica', null, 5100, 1024);
						this.describeAColumn(table, 'strutturakind_title', 'Tipologia Tipo Struttura didattica', null, 5220, 50);
						this.describeAColumn(table, 'esonero_titolostudio_nellistituto', 'Solo per corsi dell\'istituto', null, 6000, null);
						this.describeAColumn(table, 'costoscontodef_title', 'Sconto', null, 6200, 2024);
						this.describeAColumn(table, 'esonero_titolostudio_noabbr', 'Senza abbreviazioni di carriera', null, 7000, null);
						this.describeAColumn(table, 'esoneroanskind_title', 'Tipologia Codice ANS', null, 7200, 50);
						this.describeAColumn(table, 'esoneroanskind_description', 'Descrizione Codice ANS', null, 7300, 256);
						this.describeAColumn(table, 'esonero_retroattivo', 'Retroattivo', null, 8000, null);
						this.describeAColumn(table, 'esonero_titolostudio_noparttime', 'Senza aver effettuato iscrizioni part-time', null, 8000, null);
						this.describeAColumn(table, 'esonero_soloconisee', 'Applicabile solo con ISEE', null, 9000, null);
						this.describeAColumn(table, 'esonero_titolostudio_votomin', 'Voto minimo', null, 9000, null);
//$objCalcFieldConfig_titolostudio$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idesonero"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "titolostudio": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('esonerotitolostudioview', new meta_esonerotitolostudioview('esonerotitolostudioview'));

	}());
